packages/core/src/dialects/abstract/connection-manager.ts

```diff
@@ -202,6 +202,10 @@ export class AbstractConnectionManager<TConnection extends Connection = Connecti
   }
 
   async _initDatabaseVersion(conn?: TConnection) {
+    if (this.sequelize.dialect.name === 'momento') {
```

Comment:
this seems fine for now, but just to make sure I understand this code, if we wanted to avoid the leaking of dialect details here, we would probably just need to implement a few more functions such as `connect` and `fetchDatabaseVersion` in our dialect and have them implemented as no-ops?

Not asking for any changes for now, just scouting the path forward if we were to try to make an official PR at some point.

Reply:
This one we can probably avoid leaking but for some of things such as `define` the only way to avoid leaking would be to extend the `Sequelize` class and ask users to instantiate a `MomentoSequelize` instead. I might be wrong though and have not given a very deep look - but did dig into it a bit and came to this conclusion.